src/iq.rs

```diff
@@ -307,6 +307,9 @@ impl IQClient {
             Err(e) => return Err(Box::new(e)),
         };
 
+        if internal_application_id.applications.len() == 0 {
+            return Err(Box::new(GeneralError(format!("could not get internal application id for public application id: {}. perhaps you lack permissions on this application?", app.to_string()).into())));
```

Comment:
We could probably clean this up using [`eyre`](https://crates.io/crates/eyre) or [`anyhow`](https://crates.io/crates/anyhow) with something like:

``` rust
bail!("could not get internal application id for public application id: {}. perhaps you lack permissions on this application?", app.to_string())
```

and changing the return to `eyre::Result<ReportResults>`

This could probably be done in a separate PR, but I figured it was worth mentioning.

Reply:
Very interesting stuff. Thanks for the suggestion. Not sure I grok it all well 'nuf to add to this PR right now.